### ISSUE TITLE
DB4S (v3.11.2, X64) Missing api-ms-crt-runtime

### ISSUE THREAD

**Issue by ArtKns** [sqlitebrowser/sqlitebrowser#1954]
Details for the issue
--------------------

#### What did you do?
Clean install DB4S X64 v3.11.2 (Win 7, X64 Home/MSI Installer)
Installed on local non-windows (non-systems) HDD in user specified dir
   (I.E., not in %programfiles%)
DB Browser for SQLite.exe  4,257,792  4-03-19 14:13:24

#### What did you expect to see?
A running application

#### What did you see instead?
Run time failure Error Message:
"The program can't start because api-ms-crt-runtime-I1-1-0.dll
   is missing from your computer. 
Try reinstalling the program to fix this problem."

Useful extra information
-------------------------
Tested previously installed v3.6.0 X32 (Win 7, X64 Home)
sqlitebrowser.exe 1,836,544 5-05-15  9:49:56
Start Failed: Error message =
"The program can't start because MCVCR120.dll
   is missing from your computer. 
Try reinstalling the program to fix this problem."
==================
This issue is not new!
From Jan 21, 016 notes:
Fresh install of X32 v3.6.0 on Win XP Pro
Start Failed: Error Message =
Run time error: Entry point not found
(I apparently did not record the entry point label)
The info below often helps, please fill it out if you're able to. :)

Attached is the MSI installation report generated by a NirSoft utility

#### What operating system are you using?
- [x] Windows: ( 7 version: Home )
- [ ] Linux: ( _distro:_ ___ )
- [ ] Mac OS: ( _version:_ ___ )
- [ ] Other: ___

#### What is your DB4S version?
- [x] 3.11.2
- [ ] 3.11.1
- [ ] 3.10.1
- [ ] Other: ___

#### Did you also
- [ ] Try out the latest nightly version: https://github.com/sqlitebrowser/sqlitebrowser#nightly-builds
- [x] Search for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by ArtKns** [comment]
Sorry about the shouting - don't know what happened. Newbie to Github here. Didn't see the preview button and text had looked OK when written but now I can't edit the body. 

I also don't see the attached file ref anywhere  so will try again.

[DB Browser v3_11_2 (nirsft,T404,X64)-T801 DB_MSI Install Info+Files.txt](https://github.com/sqlitebrowser/sqlitebrowser/files/3459278/DB.Browser.v3_11_2.nirsft.T404.X64.-T801.DB_MSI.Install.Info%2BFiles.txt)

Ah. Looks like it worked this time.

**Comment by karim** [comment]
Hello, and sorry for the late reply.

This issue seems to be related to a missing Visual C++ Redistributable from your system. Please check this comment in the following issue https://github.com/sqlitebrowser/sqlitebrowser/issues/1915#issuecomment-515873463 for more details.

DB Browser for SQLite depends on Visual C++ 2015 Redistributable for the 32-bit version, and Visual C++ 2017 Redistributable for the 64-bit version. We bundle them already with our installer/zip packages.

However, in the issue I linked above we found out that one of our dependencies (OpenSSL library) also depends on Visual C++ 2013 Redistributable, so you need that installed as well. We haven't figured the best way to solve this problem yet. Should we bundle it as well, or re-build OpenSSL ourselves.

In the meantime, please install Visual C++ 2013 Redistributable (https://www.microsoft.com/en-eg/download/details.aspx?id=40784) yourself to fix the missing DLL error.

Sorry! :smiley:

**Comment by scottfurry** [comment]
@karim - this sounds like something that _should be_ rebuilt.
Having end users install three different IDE versions to run DB4S seems a touch harsh, IMHO.
32/64-bit issue. Okay. One version for each. But we should consider(file a feature request so we don't lose track), "Rebuild OpenSLL for VS version" task.

**Comment by karim** [comment]
> @karim - this sounds like something that _should be_ rebuilt.

I wholeheartedly agree.

However, building OpenSSL on Windows seems to be a complicated task, judging by a quick search (https://gist.github.com/terrillmoore/995421ea6171a9aa50552f6aa4be0998). I stopped reading halfway through. :smiley: I will give it another try when I have some free time.

> But we should consider(file a feature request so we don't lose track), "Rebuild OpenSLL for VS version" task.

You are right, I will do that. Maybe we can close other open issues too and direct all new ones to it.

**Comment by scottfurry** [comment]
> However, building OpenSSL on Windows seems to be a complicated task, judging by a quick search (https://gist.github.com/terrillmoore/995421ea6171a9aa50552f6aa4be0998). I stopped reading halfway through.  I will give it another try when I have some free time.

Be afraid not of the compiler, young padawan.
We'll sort it out.

> > But we should consider(file a feature request so we don't lose track), "Rebuild OpenSLL for VS version" task.
> 
> You are right, I will do that. Maybe we can close other open issues too and direct all new ones to it.

I don't have an exact number, but I can recall a dozen or so bug reports of this nature all making noise about missing redistributable DLL. If OpenSSL is the problem, let's make that go away with a sense of finality. Future *US* will be grateful when next version of VS comes out. _There can be only one__

**Comment by ArtKns** [comment]
At 08-02-2019 16:01, Karim ElDeeb wrote:
>Hello, and sorry for the late reply.
What??? A reply in less than 24 hrs is hardly late!
Thank you for your quick reply.

>This issue seems to be related to a missing Visual C++ 
>Redistributable from your system. Please check this comment in the 
>following issue 
><https://github.com/sqlitebrowser/sqlitebrowser/issues/1915#issuecomment-515873463>#1915 
>(comment) for more details.
>DB Browser for SQLite depends on Visual C++ 2015 Redistributable for 
>the 32-bit version, and Visual C++ 2017 Redistributable for the 
>64-bit version. We bundle them already with our installer/zip packages.
>However, in the issue I linked above we found out that one of our 
>dependencies (OpenSSL library) also depends on Visual C++ 2013 
>Redistributable, so you need that installed as well. We haven't 
>figured the best way to solve this problem yet. Should we bundle it 
>as well, or re-build OpenSSL ourselves.

Neither VS 2013 Redist nor VS 2015 Redist were installed. I installed 
both but  api-ms-win-crt-runtime-l1-1-0.dll was still missing. I 
think the problem is not with the Redists but  with my configuration. 
It seems that I have never installed Win 7 SP1. The VS 2015 Redist 
installer could not find it and failed.

Unfortunately, at this time I am unable to complete an installation 
of Win 7 SP1 as I am leaving for New York in a couple of weeks and 
time is short. SP1 wants 8GB free space and there is only about 2GB 
left on the Windows partition. I gave it a dedicated 16GiB partition 
(4x its initial size), install all apps on other disks but Windows 
has filled the disk with creeping crud. In their great "wisdom" MS 
stores the installation files for each and every appl, driver, 
whatever and for every version in the Winsxs directory. ^GB of 
useless garbage that can't be moved or deleted as it is full or 
reparse points. When I get back I will work on this.

Art

**Comment by karim** [comment]
Have you tried using the .zip version of DB Browser for SQLite? Maybe the installer is having trouble installing the missing .dll correctly (inside system folder). The .zip version should have all the required dlls in the same directory.

**Comment by ArtKns** [comment]
At 08-05-2019 12:35, Karim ElDeeb wrote:
>Have you tried using the .zip version of DB Browser for SQLite? 
>Maybe the installer is having trouble installing the missing .dll 
>correctly (inside system folder). The .zip version should have all 
>the required dlls in the same directory.

No, I hadn't. I didn't realize that DB4S was portable. That would 
have saved me a day and a half.

The zip extraction installation works like a champ and is SO much 
easier (and, it turns out, much more reliable). The MSI procedure and 
all its hassle, side effects (a la \winsxs), and special requirements 
seems to me a super high cost to pay just to autogenerate a couple of 
links. I would suggest you clarify or emphasize that on the download 
page. I am sure it would save a lot of people (including you as a 
support team), a lot of time and effort.

Thank you very much for your speedy and effective response.

Art

**Comment by ArtKns** [comment]
In light of the hassle, problems, wasted time and effort caused by 
the MSI installer and how complete and easy it was to install DB4S 
from the zip distribution I really have to question why you bother 
with a MSI installer at all. It really adds nothing but hassle and 
appears to now require a major effort to accommodate missing files 
and multiple redistribution packages.

Why not just get rid of the entire MSI process and just publish the 
zip file with instructions on creating desktop and/or Start Menu 
links? If you absolutely need an executable distribution, package 
DB4S as a self-extracting archive with a small standard app to create 
the links. In other words, the KISS method.

Art

**Comment by chrisjlocke** [comment]
I think while DB4S is quite a 'high end' product (ie, I would assume used
by computer literate people) it was required that it has a very simple*
installation method.  MSI installers seemed the best way to go.

*In an ideal world, simply double clicking the installer would do the
needful.

>

**Comment by karim** [comment]
> I didn't realize that DB4S was portable.

It's not *really* portable as it saves its settings in Windows Registry. It's actually created by extracting the content of MSI installer. Something like..

> msiexec.exe /a "installer.msi" /q TARGETDIR=C:\path

We want to create a true portable version where the settings are saved in .ini file in the same directory. It's not a high priority at the moment.

> That would have saved me a day and a half.

Sorry for wasting your time but we were trying to find the root cause of the problem. :smiley: 

This is the first time we encounter this issue where the installer is unable to install Visual C++ Redist on Windows 7 (without SP1). We should probably add a condition to stop installation in this case with a helpful message. An alternative solution is to statically link the runtime library.

**Comment by ArtKns** [comment]
At 08-06-2019 06:55, Karim ElDeeb wrote:

>I didn't realize that DB4S was portable.
>
>It's not really portable as it saves its settings in Windows 
>Registry. It's actually created by extracting the content of MSI 
>installer. Something like..
>
>msiexec.exe /a "installer.msi" /q TARGETDIR=C:\path
>
>We want to create a true portable version where the settings are 
>saved in .ini file in the same directory. It's not a high priority 
>at the moment.

Yes and no. I noticed the entries in the registry but they were all 
created at run time and didn't affect the configuration in any way. 
Yes, a truly portable app would not use the registry at all but for 
all practical purposes it is a "portable" desktop application. To me, 
an extract and run application that does not store any config info in 
the registry is portable. I probably should have put my original 
comment in quotes.


>That would have saved me a day and a half.
>Sorry for wasting your time but we were trying to find the root 
>cause of the problem.

Hey, no problem! It would not have taken very long to just install 
the redist packages. It was my choice and curiosity curse to update 
and try other packages and appls to see how they were affected. I 
wasn't blaming you. Remember you did solve the problem by suggesting 
installing the zipped version.

I do think that a short note regarding the simplicity of using the 
zip file and what little registry usage transpires would be very 
helpful no matter what you do and I think it  would preclude any need 
for a truly portable version.

**Comment by ArtKns** [comment]
At 08-06-2019 06:05, Chris Locke wrote:
>I think while DB4S is quite a 'high end' product (ie, I would assume used
>by computer literate people) it was required that it has a very simple*
>installation method. MSI installers seemed the best way to go.
>
>*In an ideal world, simply double clicking the installer would do the
>needful.

No matter what the level of literacy or geekiness, a single click 
install is appreciated.

However, that does not necessitate an MSI installer. As I stated 
there are tremendous downsides to the MSI installer. A simple 
self-extracting zip file with  code to create desktop and/or start 
menu links would still be a single click solution and save everybody 
time and effort.

Art

**Comment by karim** [comment]
I forgot to mention that we do have another version (other than the .msi and .zip) that can be considered portable. If you checked the releases section https://github.com/sqlitebrowser/sqlitebrowser/releases/latest you will see another .exe file. It is a https://portableapps.com/ version.

While the .zip version is working well for you, some users might find the .msi useful. DB Browser for SQLite is known to be used by online courses (teaching SQL), and even governments, where an administrative mass and/or silent install on a lot of PCs is preferred. 

Finally, numbers don't lie. :smiley: 

We (actually Justin :smile:) are running a stats server (https://sqlitebrowser.org/stats/) to monitor the downloads and so far the (64-bit) .msi is winning. :smile:

**Comment by ArtKns** [comment]
At 08-08-2019 11:49, Karim ElDeeb wrote:
>I forgot to mention that we do have another version (other than the 
>.msi and .zip) that can be considered portable. If you checked the 
>releases section 
><https://github.com/sqlitebrowser/sqlitebrowser/releases/latest>https://github.com/sqlitebrowser/sqlitebrowser/releases/latest 
>you will see another .exe file. It is a 
><https://portableapps.com/>https://portableapps.com/ version.
Yes. I saw that but I was not interested burning a CD.

>While the .zip version is working well for you, some users might 
>find the .msi useful. DB Browser for SQLite is known to be used by 
>online courses (teaching SQL), and even governments, where an 
>administrative mass and/or silent install on a lot of PCs is preferred.

No argument that a single click install is desired but I will say 
again (since you seem to be completely ignoring my suggestion), an 
MSI installer is NOT the only possibility available nor, IMNOHO, is 
it the most desirable. A simple self extracting zip file can be a 
single click installer with the addition of a standard utility to 
create your desktop and/or start menu shortcuts. This avoids the 
entire problem of having to install multiple (changing) C++ 
redistribution packages, developer time updating and maintaining MSI 
installer packages, garbage accumulation under windows\winsxs, 
support time such as this when things go wrong to list just a few. 
There are thousands upon thousands of SW distributions that do that 
quite successfully today. Not everyone is beholden to the MS behemoth 
and their inefficient way of doing things.

>Finally, numbers don't lie.

Ah, but they do! Especially when you tout numbers comparing apples and oranges.

>We (actually Justin) are running a stats server 
>(<https://sqlitebrowser.org/stats/>https://sqlitebrowser.org/stats/) 
>to monitor the downloads and so far the (64-bit) .msi is winning.
Sure. But those numbers are meaningless. You only offer two options 
and hide what they really are.

If I offered people a choice between a memorable experience or some 
food a significant number of people would choose experience. If I 
disclosed the the "experience" was a poke in the eye and the food was 
ice cream what do you think would happen. If I offered a choice 
between cake and two different flavors of ice cream the numbers would 
be more realistic and more distributed.

While not exactly portable, if you just added a statement that the 
zip file is simply "Extract and run" I would bet that you would see a 
big uptake on the zip downloads. To give a truer picture, clarifying 
that the zip file is completely self contained while the msi file 
requires multiple additional packages would shift the numbers even 
more. Make the zip self-extracting as discussed above and I would bet 
dollars to donuts that the msi package would soon be left in the dust.

Since adding such a statements to the download web page is so 
trivial, I challenge you to do it. I will really wonder why if you don't.

I think I have beaten this issue to death. Thank you again for your 
help and good luck with DB4S.

Art

**Comment by justinclift** [comment]
> A simple self-extracting zip file with  code to create desktop and/or start
menu links would still be a single click solution and save everybody
time and effort.

Yeah, been there and done that before.  It does work ok for a lot of situations, but we moved to the .msi installer approach for Windows as that's the recommended/best-practise approach by the OS vendor.

> Since adding such a statements to the download web page is so trivial, I challenge you to do it. I will really wonder why if you don't.

We started with an [CMake](https://cmake.org) + [NSIS](https://nsis.sourceforge.io/Main_Page) based .exe approach, then a while ago moved to the .msi installer version you see.  The .msi installer version allowed us to also offer a decent .zip, as it allows unpacking the msi merge modules so we don't need to manually include the MSVC redistributables.  We recently found out we still need to update our .msi (and thus the .zip) to include some more pieces. :frowning: 

With the .zip file version, it does seem to suit some people well.  But the call for offering it was just an occasional request over the years, so I wouldn't really term it as "strong demand".

For future direction though, who knows.  We may go down that road, but (to me) at the moment it feels like it'd be a step back.  Most people seem happy with the .msi, and we've managed to get that working well enough.

> I think I have beaten this issue to death. Thank you again for your help and good luck with DB4S.

You're welcome Art. :smile:

**Comment by chrisjlocke** [comment]
> Yes. I saw that but I was not interested burning a CD.

Where did you get 'burn a CD' from?  PortableApps is a well respected software 'platform' that bundles a 'runner' application, helpfully making DB4S fully portable.  
DB4S (for the Windows version, anyway) writes to the registry, so this runner exports this to a small file on exit, and reads it back in when you start.  Just to clarify, it has nothing to do with CDs.

**Comment by ArtKns** [comment]
At 08-11-2019 00:22, Chris Locke wrote:

>Yes. I saw that but I was not interested burning a CD.
>
>Where did you get 'burn a CD' from? PortableApps is a well respected 
>software 'platform' that bundles a 'runner' application, helpfully 
>making DB4S fully portable.
>DB4S (for the Windows version, anyway) writes to the registry, so 
>this runner exports this to a small file on exit, and reads it back 
>in when you start. Just to clarify, it has nothing to do with CDs.

Actually, it did.

PortableApps was originally designed to facilitate application 
installs that did not rely on the windows registry. This then 
permitted running collections of said installations from a CD. There 
was also a mechanism for setting up a application launch menu and on 
(CD) disc tree structure. This required a fair number of additional 
configuration and ini files over and above the application's 
requirements. The launcher required the application to be installed 
under the launcher's path. It did not facilitate installing a 
"portable app" in an arbitrary location on a system and with a 
standard desktop/start menu shortcut link.

That is what I was alluding to with my post. "PortableApps" is NOT 
the same thing as "portable apps".

Art

**Comment by ArtKns** [comment]
At 08-10-2019 23:41, Justin Clift wrote:

>A simple self-extracting zip file with code to create desktop and/or start
>menu links would still be a single click solution and save everybody
>time and effort.
>
>Yeah, been there and done that before. It does work ok for a lot of 
>situations, but we moved to the .msi installer approach for Windows 
>as that's the recommended/best-practise approach by the OS vendor.

Of course it is! In what universe would Microsoft recommend a 
non-Microsoft solution!

>Since adding such a statements to the download web page is so 
>trivial, I challenge you to do it. I will really wonder why if you don't.
>
>We started with an <https://cmake.org>CMake + 
><https://nsis.sourceforge.io/Main_Page>NSIS based .exe approach, 
>then a while ago moved to the .msi installer version you see. The 
>.msi installer version allowed us to also offer a decent .zip, as it 
>allows unpacking the msi merge modules so we don't need to manually 
>include the MSVC redistributables. We recently found out we still 
>need to update our .msi (and thus the .zip) to include some more pieces.

OK. Automating the identification and collection of all required 
files into a zip file is very desirable. What I suggested was to 
start from the zip file and convert it into a self extracting SFX 
file. You could have even gone so far as to use the MSI installer 
strictly as an internal zip generator and not distribute the MSI 
installer at all. Highly unlikely given this discussion. ;-)

>With the .zip file version, it does seem to suit some people well. 
>But the call for offering it was just an occasional request over the 
>years, so I wouldn't really term it as "strong demand".
>
><snip> Most people seem happy with the .msi, and we've managed to 
>get that working well enough.

"well enough" leaves a lot of wiggle room.

Thank you for your response but you did not address a single one of 
my points. Why not? Why would you not even make a trivial text 
addition to your website to clarify the DL options? It makes me 
wonder if you are getting paid or sponsored by Microsoft!

**Comment by mgrojo** [comment]
I'm a developer of this project but my background knowledge is in Linux, not Windows. I'd dare to say that Windows packaging system sucks a lot, but I'd never say that a better alternative is having a bunch of zip files manually managed by the user. That is like saying that wearing furs is better than wearing cloths, just because there aren't of high quality in your local stores :wink:

In any case, it doesn't matter, because we provide all the options, and our users are free to use whatever they like or need in every situation. To those that don't like the Microsoft's behemoths, my recommendation is: be brave and get free of them entirely :smile:

**Comment by justinclift** [comment]
> Thank you for your response but you did not address a single one of
my points.

Huh?

Sorry, I actually thought I took time **specifically** to address your points. :wink:

Let me rephrase our packaging decisions so far:

* We're a volunteer effort, doing stuff "as we feel like it"
* The current result is the result of volunteers having felt like it
* Converting the .zip approach into an .exe may at some point be done, if someone feels like it.  I sure as heck don't atm, so it won't be me. :smile:

**Comment by justinclift** [comment]
> It makes me wonder if you are getting paid or sponsored by Microsoft!

Dude.  Insulting me like that is **not** going to help.  At all.

**Comment by karim** [comment]
Hello Art,

> No argument that a single click install is desired but I will say again (since you seem to be completely ignoring my suggestion), an MSI installer is NOT the only possibility available nor, IMNOHO, is it the most desirable.

No one is ignoring your suggestion, and this issue is still open so we can discuss it further.

It is just that your suggestion came out as; MSI is bad please drop it and the .zip should be enough for all of your users, or if they needed an installer give them the same .zip as a self-extracting archive.

If your suggestion is about completely dropping the MSI installer, then it is not going to happen. Simply because some of our users are using it. Why are they using it? It's up to them. We even support Windows XP even though a lot of people (including in our team) are saying we shouldn't.

However, if your suggestion was about providing a self-extracting archive along with, or as an alternative to, the installer then it is doable. All you have to do is open a new issue so we can keep track of it, and we will work on it when we have the time.

I'm sure it is something trivial, or a simple command to create one from a .zip file. If you know how it's done you can open a PR and we will happily merge it.

**Comment by karim** [comment]
> This avoids the entire problem of having to install multiple (changing) C++ redistribution packages, developer time updating and maintaining MSI installer packages, garbage accumulation under windows\winsxs, support time such as this when things go wrong to list just a few.

No, it doesn't. The redistribution packages must exist for the application to function. We simply opt to install them in the system folder so:

- They can be updated by the OS, since it is one of its components
- They can be skipped if they are already there (probably the same or better version)

For this particular issue we could do one of the following:

1. Link it statically to the .exe (not recommended since it can no longer be updated by the OS)
2. Install it in the application directory (also not recommended for the same reason)
3. Install it in the application directory *only* if we can't install in the system directory (Win7 with no SP)

The third option should fix this open issue and allow you to use the installer without a problem.

**Comment by karim** [comment]
> Ah, but they do! Especially when you tout numbers comparing apples and oranges.

I was comparing the installer versus the .zip archive, and it seems people prefer to use the installer. I think we both can agree on that. In fact, most people who use the .zip archive do so because they lack admin privilege on their (work or study) PC.

If you meant that I should be comparing the self-extracting archive (.exe) versus the MSI installer (.msi) since both are installers, then I don't see how this is going to help our users.

More often than not, it will confuse them as they have to decide now which installer is better, while what they really want is to use the application.

This is akin to saying we should drop the .zip archive and use .7z (7zip), or .rar (WinRAR) archive, because it provides a better compression ratio.

It will be a maintenance nightmare (to provide both) as we have to handle situations where a user installs DB4S with the .msi installer, then upgraded it with the .exe installer, and nothing is working anymore.

**Comment by karim** [comment]
> Sure. But those numbers are meaningless. You only offer two options and hide what they really are.

Hide what exactly?

The Windows section in our readme (https://github.com/sqlitebrowser/sqlitebrowser#windows) provides a link (https://sqlitebrowser.org/dl/#windows) for our Windows downloads that clearly shows what they are.

If you are using the nightly (https://nightlies.sqlitebrowser.org/latest/), there is also a readme file (https://nightlies.sqlitebrowser.org/latest/%2A%20README.txt) there.

> While not exactly portable, if you just added a statement that the zip file is simply "Extract and run" I would bet that you would see a big uptake on the zip downloads.

As opposed to the installer being "Crash and burn"?

I could also add that the installer will increase your laptop battery life because extracting the .zip archive will run some expensive decompression algorithms that will waste CPU cycles in order to skew the stats in favor of the installer.

It is not a contest. We provide both and users are free to pick the one they prefer.

> To give a truer picture, clarifying that the zip file is completely self contained while the msi file requires multiple additional packages would shift the numbers even more.

That would be wrong since both are self contained. The .zip is actually created out of the .msi file.

>  Make the zip self-extracting as discussed above and I would bet dollars to donuts that the msi package would soon be left in the dust.

No, it will not.

A self-extracting archive does nothing more than, well, extracting an archive, so you can double-click the file instead of *right-click > extract*. Do you think this provides a better UX than an installer?

At some point, you will want more from it (desktop shortcuts, file association, Registry manipulation, OS detection, logging, custom actions/dialogs, etc.) and you will find it lacking, so you will turn to scriptable installers.

> Since adding such a statements to the download web page is so trivial, I challenge you to do it. I will really wonder why if you don't.

There's no need for a challenge, and no need to wonder also. I will not do it simply because what you want us to say is wrong.

**Comment by karim** [comment]
> It makes me wonder if you are getting paid or sponsored by Microsoft!

Out of all the people on our team you picked Justin to accuse him of being paid/sponsored by Microsoft. That is really funny. :smile: 

Actually, Justin was the one who wanted us to move out of GitHub right after Microsoft acquisition. He wanted us to find another service like GitHub or be self-hosted. It was turned down by us. While we understand his reasons, we decided that the move will mostly affect our users, and so we decided to stay.

We managed to move the version check/downloads to our own servers (which costs us money) though.

Finally..

I think your confusion comes from not knowing the project history. If you have some free time, please take a look around our issues and you will find out how things evolved and came to be.

You will notice that our decisions are based on what our users want (or expect), our experience (what we have tried before and worked or didn't work), and not alienating our users. We are still jumping through hoops to make DB4S works on Windows XP.
